File: common/src/main/java/dev/niostone/interstellar/config/builder/PlanetConfigBuilder.java
package dev.niostone.interstellar.config.builder;

import dev.niostone.interstellar.config.GasConfig;
import dev.niostone.interstellar.config.PlanetConfig;
import net.minecraft.resources.ResourceLocation;

public class PlanetConfigBuilder {
    private PlanetConfig config;

    public PlanetConfigBuilder(String name) {
        this.config = new PlanetConfig(name);
    }

    public PlanetConfigBuilder(ResourceLocation name) {
        this.config = new PlanetConfig(name.toString());
    }

    public PlanetConfigBuilder(String namespace, String id) {
        this.config = new PlanetConfig(namespace + ":" + id);
    }

    public PlanetConfigBuilder enable(boolean inp) {
        this.config.enable = inp;
        return this;
    }

    public PlanetConfigBuilder dimension(int inp) {
        this.config.dimensionId = inp;
        return this;
    }

    public PlanetConfigBuilder withPressure(float inp) {
        this.config.pressure = inp;
        return this;
    }

    public PlanetConfigBuilder withRadiation(float inp) {
        this.config.radiation = inp;
        return this;
    }

    public PlanetConfigBuilder tier(int inp) {
        this.config.tier = inp;
        return this;
    }

    public PlanetConfigBuilder oxygen(boolean inp) {
        this.config.enableOxygen = inp;
        return this;
    }

    public PlanetConfigBuilder pressure(boolean inp) {
        this.config.enablePressure = inp;
        return this;
    }

    public PlanetConfigBuilder radiation(boolean inp) {
        this.config.enableRadiation = inp;
        return this;
    }

    public PlanetConfigBuilder gas(GasConfig inp) {
        this.config.gases.add(inp);
        return this;
    }

    public PlanetConfig build() {
        return this.config;
    }
}

File: common/src/main/kotlin/org/stardustmodding/interstellar/api/builder/PlanetConfigBuilder.kt
package org.stardustmodding.interstellar.api.builder

import  org.stardustmodding.interstellar.impl.config.GasConfig
import org.stardustmodding.interstellar.impl.config.PlanetConfig
import net.minecraft.resources.ResourceLocation

class PlanetConfigBuilder {
    val config: PlanetConfig

    constructor(name: String) {
        this.config = PlanetConfig(name)
    }

    constructor(name: ResourceLocation) {
        this.config = PlanetConfig(name.toString())
    }

    constructor(namespace: String, id: String) {
        this.config = PlanetConfig("$namespace:$id")
    }

    fun enable(value: Boolean) = apply { this.config.enable = value }
    fun dimension(value: Int) = apply { this.config.dimensionId = value }
    fun enablePressure(value: Boolean) = apply { this.config.enablePressure = value }
    fun enableRadiation(value: Boolean) = apply { this.config.enableRadiation = value }
    fun tier(value: Int) = apply { this.config.tier = value }
    fun enableOxygen(value: Boolean) = apply { this.config.enableOxygen = value }
    fun pressure(value: Float) = apply { this.config.pressure = value }
    fun radiation(value: Float) = apply { this.config.radiation = value }
    fun gas(value: GasConfig) = apply { this.config.gases.add(value) }

    fun build(): PlanetConfig {
        return this.config
    }
}
